### PR TITLE
feat: refactor OT log collector configuration

### PR DESCRIPTION
**Pull request by swiatekm-sumo**
##### Description

Change how log collection using OT is configured. The idea here is for the purely functional configuration to live under the `sumologic.logs` key and contain no application-specific configuration, and for the K8s specific and application specific (like raw OT configuration) configuration to be separate - currently I've left it under `otellogs`, but I'm open to alternatives.

TODO:
- [x] Add template tests for the new options
- [ ] setting for tailling additional files
- [x] documentation
---

##### Checklist

- [x] Changelog updated

###### Testing performed

- [X] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: deploy/helm/sumologic/conf/logs/collector/otelcol/config.yaml
Comment (sumo-drosiek) [inline deploy/helm/sumologic/conf/logs/collector/otelcol/config.yaml:131]: I would reconsider supporting multiple mutltiline detections in this PR as this is common use case for customers. I can also create Issue if we prefer to add it in the future

File: deploy/helm/sumologic/conf/logs/collector/otelcol/config.yaml
Comment (sumo-drosiek) [inline deploy/helm/sumologic/conf/logs/collector/otelcol/config.yaml:200]: remove operator is already supported

```suggestion
      - type: remove
        field: attributes["log.file.path"]
```

File: deploy/helm/sumologic/templates/_helpers.tpl
Comment (sumo-drosiek) [inline deploy/helm/sumologic/templates/_helpers.tpl:1356]: We should have possibility to disable otellogs even if logs are enabled

File: deploy/helm/sumologic/values.yaml
Comment (sumo-drosiek) [inline deploy/helm/sumologic/values.yaml:4867]: I would leave this option

File: tests/integration/values/values_helm_otelcol_logs.yaml
Comment (sumo-drosiek) [inline tests/integration/values/values_helm_otelcol_logs.yaml:46]: I like it :)

**Review by sumo-drosiek — COMMENTED** [review]
Overall this is fine. Are we going to document whose changes in this PR or separate one?

**Comment by astencel-sumo** [comment]
I believe we should start with the documentation (this PR doesn't add or modify any). Writing the documentation should help us make sure the configuration properties that we define in `values.yaml` cover the scenarios we want to cover and that the names/structure make sense.

File: deploy/helm/sumologic/values.yaml
Comment (swiatekm-sumo) [reply deploy/helm/sumologic/values.yaml:4911]: Moved to `sumologic.logs.collector`.

File: deploy/helm/sumologic/conf/logs/collector/otelcol/config.yaml
Comment (swiatekm-sumo) [reply deploy/helm/sumologic/conf/logs/collector/otelcol/config.yaml:132]: I think that should be added in a separate PR. Sounds like a complicated use case.

File: deploy/helm/sumologic/conf/logs/collector/otelcol/config.yaml
Comment (swiatekm-sumo) [reply deploy/helm/sumologic/conf/logs/collector/otelcol/config.yaml]: Done.

File: deploy/docs/opentelemetry_collector.md
Comment (sumo-drosiek) [inline deploy/docs/opentelemetry_collector.md:127]:
```suggestion
SystemD logs are collected by default. This can be disabled by setting:
```

**Review by sumo-drosiek — APPROVED** [review]
LGTM, please get approval from someone else as well :)

File: deploy/docs/opentelemetry_collector.md
Comment (astencel-sumo) [inline deploy/docs/opentelemetry_collector.md:55]: Micro-nit: I think it's "Fluentd", not "FluentD" :)

Comment (astencel-sumo) [reply deploy/docs/opentelemetry_collector.md:55]: Also I believe Fluent Bit does not use a hyphen in the name: https://fluentbit.io/
image

File: deploy/docs/opentelemetry_collector.md
Comment (astencel-sumo) [inline deploy/docs/opentelemetry_collector.md:86]:
```suggestion
[`values.yaml`][values] under the `sumologic.logs` key.
```

File: deploy/docs/opentelemetry_collector.md
Comment (astencel-sumo) [inline deploy/docs/opentelemetry_collector.md:88]:
```suggestion
Configuration specific to the log collector DaemonSet can be found under the `otellogs` key.
```

File: deploy/docs/opentelemetry_collector.md
Comment (astencel-sumo) [inline deploy/docs/opentelemetry_collector.md:90]:
```suggestion
Finally, configuration specific to the metadata enrichment StatefulSet can be found under the `metadata.logs` key.
```

File: deploy/helm/sumologic/values.yaml
Comment (astencel-sumo) [inline deploy/helm/sumologic/values.yaml:226]: What do you think about the name `first_line_regex` instead of `start_regex`? I feel that "start" is a bit more ambiguous to me.

Also is the `.*` at the end of the regex required?

File: deploy/docs/opentelemetry_collector.md
Comment (astencel-sumo) [inline deploy/docs/opentelemetry_collector.md:102]:
```suggestion
Multiline log parsing for OpenTelemetry Collector can be configured using the `sumologic.logs.multiline` section in [values.yaml][values].
```

File: deploy/docs/opentelemetry_collector.md
Comment (astencel-sumo) [inline deploy/docs/opentelemetry_collector.md:127]: Again, I don't think there's a capital "D" in "systemd".
```suggestion
Systemd logs are collected by default. This can be disabled by setting:
```

File: deploy/helm/sumologic/values.yaml
Comment (sumo-drosiek) [reply deploy/helm/sumologic/values.yaml:226]: > Also is the .* at the end of the regex required?

it is not afaik

File: deploy/docs/opentelemetry_collector.md
Comment (swiatekm-sumo) [reply deploy/docs/opentelemetry_collector.md:102]: This is actually wrong elsewhere in the document as well, I can fix it while I'm at it.

File: deploy/helm/sumologic/values.yaml
Comment (swiatekm-sumo) [reply deploy/helm/sumologic/values.yaml:226]: > What do you think about the name `first_line_regex` instead of `start_regex`? I feel that "start" is a bit more ambiguous to me.

Sounds good to me, I didn't really put much thought into the name of this option. I don't think there's any standard terminology for this.

File: deploy/helm/sumologic/values.yaml
Comment (astencel-sumo) [inline deploy/helm/sumologic/values.yaml:231]: Correct me if I'm wrong, but it looks like there's now no simple way to set e.g. a custom source category on logs from systemd?

It's just an observation - I don't think it's a blocker for this PR, as this can be added in a later PR, right?

File: deploy/helm/sumologic/values.yaml
Comment (swiatekm-sumo) [reply deploy/helm/sumologic/values.yaml]: Done.

File: deploy/helm/sumologic/values.yaml
Comment (swiatekm-sumo) [reply deploy/helm/sumologic/values.yaml:231]: That's something that the metadata layer does, so we'd have to add it there, ideally while also moving configuration to a template like I've done here. This PR is only concerned with configuration for the log collector.

File: deploy/docs/opentelemetry_collector.md
Comment (astencel-sumo) [inline deploy/docs/opentelemetry_collector.md:112]:
```suggestion
where `first_line_regex` is a regular expression used to detect the first line of a multiline log.
```

File: deploy/helm/sumologic/templates/_helpers.tpl
Comment (astencel-sumo) [inline deploy/helm/sumologic/templates/_helpers.tpl:1375]: Do we really want to explicitly forbid the combination of both OT logs collector and Fluent Bit?

I can imagine a user might e.g. want to collect container logs with OT and systemd logs with FB (or the other way around) for some reason and I'm not convinced that hardcoding the lack of this possibility is a good idea.

Comment (sumo-drosiek) [reply deploy/helm/sumologic/templates/_helpers.tpl:1375]: I think that customers more likely forget to disable one of the collector during migration than will try to use both of them. Especially when we deprecate fluent-bit

Comment (swiatekm-sumo) [reply deploy/helm/sumologic/templates/_helpers.tpl:1375]: I think the value of catching accidental misconfigurations is greater than the value of allowing users to have both simultaneously. Now that OT supports systemd logs, I don't think there's any reason to run them side-by-side.

File: deploy/helm/sumologic/templates/logs/collector/otelcol/daemonset.yaml
Comment (astencel-sumo) [inline deploy/helm/sumologic/templates/logs/collector/otelcol/daemonset.yaml:80]: So it is now not possible to customize the path to OTC storage? We can add this option later though, right?

Comment (swiatekm-sumo) [reply deploy/helm/sumologic/templates/logs/collector/otelcol/daemonset.yaml:80]: I used to have a separate setting for this, but I ultimately decided it didn't add any value, and only made things more confusing in the context of the log collector, where there's no persistent queue, the volume is a hostPath, and it's only used for storing the filelog receiver state.

For metadata, I think this configuration can be moved to `sumologic.logs`, like in #2444 .

File: deploy/docs/opentelemetry_collector.md
Comment (swiatekm-sumo) [reply deploy/docs/opentelemetry_collector.md]: Fixed

File: deploy/helm/sumologic/values.yaml
Comment (perk-sumo) [inline deploy/helm/sumologic/values.yaml:222]: Can we change the name of this key to make it obvious it is about otelcol?
Current name would be OK in v4 when we'll have only single agent type but right now I think we should be more explicit.

I suggest make it:
```
sumologic:
  logs:
    collector:
      otelcol:
        enabled: true
```

Comment (swiatekm-sumo) [reply deploy/helm/sumologic/values.yaml]: Done.

File: deploy/helm/sumologic/templates/logs/collector/otelcol/daemonset.yaml
Comment (astencel-sumo) [reply deploy/helm/sumologic/templates/logs/collector/otelcol/daemonset.yaml:80]: Fair enough, it seems there is no need to modify this path 👍

**Review by astencel-sumo — APPROVED** [review]
LGTM as long as we agree that we want to forbid the simultaneous usage of Fluent Bit and OT logs collector. I was against forbidding it, but if everyone is fine with it, I'm good.

After all, this can be changed later if we change our minds (or the customers change our minds).

File: deploy/helm/sumologic/templates/_helpers.tpl
Comment (swiatekm-sumo) [reply deploy/helm/sumologic/templates/_helpers.tpl:1375]: I added it as `allowSideBySide`, along with a documentation section.

Comment (swiatekm-sumo) [reply deploy/helm/sumologic/templates/_helpers.tpl:1375]: https://github.com/SumoLogic/sumologic-kubernetes-collection/pull/2436/commits/b7f0f80e64ce8de261f5b0417223fab4ca4038c0

File: deploy/docs/opentelemetry_collector.md
Comment (astencel-sumo) [inline deploy/docs/opentelemetry_collector.md:168]: This is true, but only when the user hasn't modified the configuration of Otelcol and Fluent Bit. How about this?
 
```
> **WARNING** Without further modifications to Otelcol and Fluent Bit configuration, this will cause each log line to be ingested twice, potentially doubling the log ingestion cost.
```

**Review by astencel-sumo — APPROVED** [review]

**Pull request merged**
